### ISSUE TITLE
Pick mic device for "hey code"

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#203640]
Testing #202926

I'd like to use the builtin macbook pro mac, instead of the airpods mic, because that puts the airpods into "mic mode" and makes audio sound bad. Is it possible to select the mic device, with the APIs that you have?

**Comment by bpasero** [comment]
Yes I share the pain of the loss in audio quality, good idea to allow to select a mic.

**Comment by ghouston** [comment]
yes, this is needed.  my machine can have 3 mics to choose from.  only one is really high quality.

**Comment by dgxo** [comment]
for now i assume vs code uses your default mic, (which you can change in settings), so until this feature is added (if ever), i would just change your default.

**Comment by bpasero** [comment]
The Azure Speech SDK that we are using allows to pick a microphone, but you need to pass in a device identifier and each platform seems to have its own way of how to obtain it.

Docs: https://learn.microsoft.com/en-us/azure/ai-services/speech-service/how-to-select-audio-input-devices

I was hoping that the following code would give me the identifier (needs [1]):

```ts
navigator.mediaDevices.enumerateDevices().then(devices => {
	devices.forEach(device => {
		console.log(device.kind + ": " + device.label + " id = " + device.deviceId);
	});
});
```

But the result I get is not something I can pass into the SDK directly it seems, at least on macOS.

However, from reading the [docs](https://learn.microsoft.com/en-us/azure/ai-services/speech-service/how-to-select-audio-input-devices) I found that on macOS you can use `BuiltInMicrophoneDevice` and that seems to point to the device microphone built-in. In my testing I can successfully listen to music in nice quality using AirPods and record my voice via `BuiltInMicrophoneDevice`. 

Maybe as a temporary fix, we provide the option to pick `BuiltInMicrophoneDevice` only on macOS until we have figured out similar things on other platforms?

[1] https://github.com/microsoft/vscode/commit/707bffbdae207740940e86a11b8178978515f9d0#diff-ce90ef6ffd304cc541efa8135dc4878c7b61889d2b2d5ef69e196af6cea7647b

**Comment by Jacksonmills** [comment]
Running VSCode from WSL2, mic works on a gaming headset microphone however when I try and set my PC to default to my Scarlett2i2 mic it still uses the gaming headset mic...

Looking forward to be able to choose or have it follow the default on Windows

**Comment by Jacksonmills** [comment]
> It should use whatever is the configured default, I am not setting a mic specifically.

In my testing when using an audio interface like the Scarlet2i2 + a boom microphone set to both default and communication default it doesn't pick it up.

When I do the exact same with a gaming headset with it's mic it works just fine.

**Comment by iscorporacion** [comment]
My case [windows] is that I have several input and output devices, but the main one is in the camera, which is very separated, then I want to use my headband, it should allow me to select the microphone I want, I will try to select it in the operating system to code predefined devices

**Comment by guenhter** [comment]
Hi, I have the same issue on Ubuntu 25.04 (but also on 24.04), that my bluetooth headset is not used whatever I do. In the System it is selected as default, but they are just not picked...

**Comment by roblourens** [comment]
Megan is implementing better voice support, and it supports this
